tests/python/pants_test/backend/jvm/tasks/jvm_compile/test_dep_exports_integration.py

```diff
@@ -0,0 +1,57 @@
+# coding=utf-8
+# Copyright 2017 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import os
+import shutil
+
+from pants.base.build_environment import get_buildroot
+from pants_test.pants_run_integration_test import PantsRunIntegrationTest
+
+
+class DepExportsIntegrationTest(PantsRunIntegrationTest):
+
+  SRC_PREFIX = 'testprojects/tests'
+  SRC_TYPES = ['java', 'scala']
+  SRC_PACKAGE = 'org/pantsbuild/testproject/exports'
+
+  @classmethod
+  def hermetic(cls):
+    return True
+
+  def test_compilation(self):
+    for lang in self.SRC_TYPES:
+      path = os.path.join(self.SRC_PREFIX, lang, self.SRC_PACKAGE)
+      pants_run = self.run_pants(['list', '{}::'.format(path)])
+      self.assert_success(pants_run)
+      target_list = pants_run.stdout_data.strip().split('\n')
+      for target in target_list:
+        pants_run = self.run_pants(['compile', '--compile-scalafmt-skip', target])
+        self.assert_success(pants_run)
+
+  def modify_exports_and_compile(self, target, modify_file):
```

Comment:
I cloned here because I want to modify the source file to test cache invalidation. I don't want to corrupt the repo, thus I copied it to a temp dir. But I will add a failure case. Benjy also mentioned it. Are you saying I don't have to run the failure test here but just add to the list you linked?